src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            [this[i], this[j]] = [this[j], this[i]];
+          }
+        } else {
+          if (String(this[i]) > String(this[j])) {
+            [this[i], this[j]] = [this[j], this[i]];
+          }
+        }
+      }
+    }
+
+    return this;
```

Comment:
The provided implementation of the sort2 method uses a basic bubble sort algorithm. Although this meets the task requirements, bubble sort has a time complexity of O(n^2), which is not efficient for large datasets. Consider implementing a more efficient sorting algorithm, like quicksort or mergesort, for better performance with large arrays.